src/server/pfs/server/driver.go

```diff
@@ -2443,6 +2523,58 @@ func nodeToFileInfo(commit *pfs.Commit, path string, node *hashtree.NodeProto, f
 	return fileInfo
 }
 
+// nodeToFileInfoInputCommit is like nodeToFileInfo, but handles the case (which
+// currently only occurs in input commits) where files have a header that is
+// stored in their parent directory
+func nodeToFileInfoInputCommit(commit *pfs.Commit, filePath string,
```

Comment:
Why not call this `nodeToFileInfoHeaderFooter` or even better just fold the logic into `nodeToFileInfo` and wrap it in a conditional.

Reply:
Also I don't think it's good to assume that only input repos will have headers, outputting stuff with headers is probably a valid usecase.

Reply:
1. Ultimately I plan to support both input and output repos with headers. I stuck with input repos only because I wanted to finish this PR in time for the release, and it's all we need for the basic version of this feature (SQL and hopefully CSV if I can finish it in time)

2. Supporting output repos in this PR would be hard (I do want to ultimately do it, but I'm trying to limit the scope here to finish the project). As for why I think it would be hard:
  - we'd have to give pipelines some way to indicate that their output has header/footer data, which isn't an interface that exists yet and we'd have to create.
  - I think folding the logic into one function would be either hard or slow—output repos don't have hashtrees that implement the hashtree interface (note the `tree` argument that this fn accepts). I could do it by having this function make a `GetFile` call, but I think that means downloading the hashtree again. By only supporting input repos, I can just pass the hashtree in.

Reply:
I'm cool with not doing this.
I do think it probably makes sense to have the function's name reference `Header/Footer` or `Shared`, since the input part seems more incidental to me. I guess it's a question of naming for where the function is used rather than what it does.

Reply:
I'm happy to change it, though to maybe add clarity to what it does, the idea is that *all* input repo nodes go through it so that the caller doesn't have to figure out or care whether the node has a header or not (it's just that output repos can't call it, and their nodes never have headers anyway, so they skip it)

Reply:
That makes sense as where it's used. I still think a name that conveys what's different about the other function of the similar name rather than where it's meant to be used conveys a little more information.

Reply:
So just to be clear I'd prefer to see the name changed here.

Reply:
Done—this is now `nodeToFileInfoHeaderFooter`